Sources/StreamChatUI/ChatChannel/ChatChannelVC.swift

```diff
@@ -243,6 +243,22 @@ open class ChatChannelVC: _ViewController,
         messageComposerVC.dismissSuggestions()
     }
 
+    open func chatMessageListVC(
+        _ vc: ChatMessageListVC,
+        decorationViewForMessage message: ChatMessage,
+        decorationType: ChatMessageDecorationType,
+        at indexPath: IndexPath
+    ) -> ChatMessageDecorationView? {
+        switch decorationType {
+        case .header where vc.shouldShowDateSeparator(forMessage: message, at: indexPath):
+            let dateSeparatorView = components.messageListDateSeparatorView.init()
+            dateSeparatorView.content = vc.dateSeparatorFormatter.format(message.createdAt)
+            return dateSeparatorView
+        default:
+            return nil
```

Comment:
Looks clean! Have you considered having separate delegate methods for the header and footer? It might make it a bit more readable as more customizations are added

Reply:
Yes this is a good point 👍 It also seems more flexible. If for some reason we need more decoration types, we will need to add new options to the enum type, which is a breaking change

Reply:
I have tried the footer/header split out in different methods a couple of times but i was reverting it back. Just couldn't make up my mind.

The only benefit that this single approach has is that for users they only have to implement/override one method instead of 2.

That being said I'm really happy that you both agree on the approach and you help choose which one to use to move forward.

@nuno-vieira more decoration types may not require addition to the enum type. Even if they do the enum is not extensible. Finally, even in the approach with the 2 methods a new decoration type will require a new method